pkg/client/aaaa.go

```diff
@@ -0,0 +1,70 @@
+package client
+
+import (
+	"context"
+	"time"
+
+	"github.com/miekg/dns"
+
+	"darvaza.org/resolver/pkg/errors"
+)
+
+var (
+	_ Client    = (*NoAAAA)(nil)
+	_ Unwrapper = (*NoAAAA)(nil)
+)
+
+// NoAAAA is a dns.Client middleware to remove AAAA entries from all responses
+type NoAAAA struct {
+	Client
+}
+
+// ExchangeContext calls the next client in the chain if it's not an AAAA,
+// and discards all AAAA entries on the response.
+func (c NoAAAA) ExchangeContext(ctx context.Context, req *dns.Msg,
+	server string) (*dns.Msg, time.Duration, error) {
+	//
+	start := time.Now()
+
+	if req == nil {
+		return nil, 0, errors.ErrBadRequest()
+	}
+
+	req.Question = trimQ(req.Question, qIsAAAA)
```

Comment:
After fighting with this for several days I am still not convinced by it, nor by my version. As far as I understand, the Question section of a dns message should be immutable so this part of the code should be something like:

```go

req2:=req.Copy()
req2.Question=trimQ(req2.Question,qIsAAAA)
if len(req2.Question)==0 {
// nothing to answer
resp:=new(dns.Msg)
resp.SetReply(req) # noty req2!!!! but the original request
return resp, time.Since(start), nil
}
```

and so forth....
basically we do not want to touch `req.Question` because that might force the client (whatever or whoever is that) to drop our answer as invalid.

The proble is that I am not yet convinced either way.

Reply:
I think this only affects `len(req.Question) > 1` hypothetical requests, but I've amended it accordingly.
I was trying to give a change to the next client to answer `req.Question[1]` of non-AAAA type